jetty-io/src/main/java/org/eclipse/jetty/io/ssl/SslConnection.java

```diff
@@ -395,22 +395,17 @@ public void failed(final Throwable x)
                     BufferUtil.clear(_encryptedOutput);
                     releaseEncryptedOutputBuffer();
 
-                    _cannotAcceptMoreAppDataToFlush = false;
-                    fail_filler = _fillRequiresFlushToProgress;
-                    if (_fillRequiresFlushToProgress)
-                        _fillRequiresFlushToProgress = false;
+                    _flushState = FlushState.IDLE;  // TODO do we need a failed state?
+                    fail_fill_interest = _fillNeedsWrap;
+                    _fillNeedsWrap = false;
                 }
 
-                failedCallback(new Callback()
+                getExecutor().execute(()->
```

Comment:
This seems like a strange change for the scope of this issue.
Why are we no longer using failedCallback()? 
Why are we dispatching a failure now?